src/MapboxStyleParser.ts

```diff
@@ -423,13 +424,26 @@ export class MapboxStyleParser implements StyleParser<Omit<MbStyle, 'sources'>>
       pitchAlignment: layout?.['icon-pitch-alignment'],
       rotate: mb2gsExpression<number>(layout?.['icon-rotate']),
       rotationAlignment: layout?.['icon-rotation-alignment'],
-      size: mb2gsExpression<number>(layout?.['icon-size']),
       textFit: layout?.['icon-text-fit'], // TODO: handle enum values
       // TODO: handle array values
       textFitPadding: layout?.['icon-text-fit-padding'] as IconSymbolizer['textFitPadding'],
       visibility: layout?.visibility && layout?.visibility !== 'none'
     };
 
+    // mabpox icon-size scales the image and does not define its size
+    if (layout?.['icon-size'] && image) {
+      const scale = mb2gsExpression<number>(layout['icon-size']) as number;
+      // mulitply the mb icon-size with the width of the sprite to get the scale
+      symbolizer.size = scale * (image.size[0] as number);
```

Comment:
Do we have to check if any of the operators is an expression and then use the geostyler `mul` expression? A simple multiplication won't work in that case, right?

Reply:
At this point the `Sprite` comes from the `getIconImage` function so it can never be a expression unless we implement it.

Reply:
But the `icon-size` and therefore also `scale` can still be an expression, right?